### ISSUE TITLE
GCE Cloudprovider `waitForOp` sometimes causes panics due to `invalid memory address or nil pointer dereference`.

### ISSUE THREAD

**Issue by saad-ali** [kubernetes/kubernetes#15978]
While investigating #15382, I discovered that the `pkg/cloudprovider/providers/gce/gce.go:waitForOp(...)` method (copied below) sometimes causes panics due to `invalid memory address or nil pointer dereference`.

```GO
func waitForOp(op *compute.Operation, getOperation func() (*compute.Operation, error)) error {
	pollOp := op
	consecPollFails := 0
	for pollOp.Status != "DONE" {
		var err error
		time.Sleep(3 * time.Second)
		pollOp, err = getOperation()
		if err != nil {
			if consecPollFails == 2 {
				// Only bail if we've seen 3 consecutive polling errors.
				return err
			}
			consecPollFails++
		} else {
			consecPollFails = 0
		}
	}
	if pollOp.Error != nil && len(pollOp.Error.Errors) > 0 {
		return &googleapi.Error{
			Code:    int(pollOp.HttpErrorStatusCode),
			Message: pollOp.Error.Errors[0].Message,
		}
	}
	return nil

}
```
Error: `E1020 20:36:34.301927    3625 util.go:82] Recovered from panic: "invalid memory address or nil pointer dereference" (runtime error: invalid memory address or nil pointer dereference)`

Stack trace indicates the culprit is this line:
```GO
	for pollOp.Status != "DONE" {
```

I'll fix this along with #15382. This method also has the potential to block indefinitely if there is a bad server response, so the fix should also fix the max amount of time to wait.

**Comment by saad-ali** [comment]
In short, anytime `getOperation()` fails `pollOp` is set to nil and `pollOp.Status` blows up. Meaning this fails everytime there is a GCE error doing a get operation call, that's pretty bad.

**Comment by saad-ali** [comment]
Marking this as required for v1.1 milestone and P0--since it affects stability of all GCE API calls.

**Comment by thockin** [comment]
status update please?

**Comment by saad-ali** [comment]
Fix is LGTM'd awaiting merge and cherry pick to 1.1

**Comment by saad-ali** [comment]
Reopening until fix is cherry-picked to v1.1

**Comment by davidopp** [comment]
Cherrypick was submitted, so closing.
